src/connector/src/state.rs

```diff
@@ -183,20 +186,17 @@ mod tests {
         }
     }
 
-    impl SourceState for TestSourceState {
-        fn identifier(&self) -> String {
+    impl SourceSplit for TestSourceState {
+        fn id(&self) -> String {
             self.partition.clone()
         }
 
-        fn encode(&self) -> Bytes {
-            let bytes = self.offset.to_le_bytes();
-            Bytes::copy_from_slice(bytes.clone().as_ref())
-            // Bytes::from(self.offset.clone().to_string())
+        fn to_string(&self) -> Result<String> {
```

Comment:
```suggestion
        fn to_json_bytes(&self) -> Result<Bytes> {
```

to_string will easily conflict with the ToString trait, besides I found that mostly Bytes is needed rather than string.

Reply:
fixed